### PR TITLE
Raise with message for 3.5.2 and earlier

### LINKED ISSUE #27247: Release 0.25.0rc0 cannot import DataFrame on Python 3.5.2
#### Code Sample, a copy-pastable example if possible
Picked up version 0.25.0rc0 last night: 
https://files.pythonhosted.org/packages/19/b6/124914598447928889c6fe08a3e40ff791e1c66c7f6e3234374eefe38359/pandas-0.25.0rc0-cp35-cp35m-win_amd64.whl#sha256=ef9976c835d4a3f1de54b0d176711d27952b48cdf752b37fe60e69395fe35b9b
```
>>>from pandas import DataFrame
...
 TypeError: descriptor '__subclasses__' of 'type' object needs an argument
```

Full traceback: 
```  
File "c:\my_project\test_runner.py", line 6, in <module>
    from pandas import DataFrame
  File "c:\my_project\venv\lib\site-packages\pandas-0.25.0rc0-py3.5-win-amd64.egg\pandas\__init__.py", line 55, in <module>
    from pandas.core.api import (
  File "c:\my_project\venv\lib\site-packages\pandas-0.25.0rc0-py3.5-win-amd64.egg\pandas\core\api.py", line 5, in <module>
    from pandas.core.arrays.integer import (
  File "c:\my_project\venv\lib\site-packages\pandas-0.25.0rc0-py3.5-win-amd64.egg\pandas\core\arrays\__init__.py", line 1, in <module>
    from .array_ import array  # noqa: F401
  File "c:\my_project\venv\lib\site-packages\pandas-0.25.0rc0-py3.5-win-amd64.egg\pandas\core\arrays\array_.py", line 7, in <module>
    from pandas.core.dtypes.common import (
  File "c:\my_project\venv\lib\site-packages\pandas-0.25.0rc0-py3.5-win-amd64.egg\pandas\core\dtypes\common.py", line 11, in <module>
    from pandas.core.dtypes.dtypes import (
  File "c:\my_project\venv\lib\site-packages\pandas-0.25.0rc0-py3.5-win-amd64.egg\pandas\core\dtypes\dtypes.py", line 53, in <module>
    class Registry:
  File "c:\my_project\venv\lib\site-packages\pandas-0.25.0rc0-py3.5-win-amd64.egg\pandas\core\dtypes\dtypes.py", line 84, in Registry
    self, dtype: Union[Type[ExtensionDtype], str]
  File "C:\Users\my_user\AppData\Local\Programs\Python\Python35\Lib\typing.py", line 552, in __getitem__
    dict(self.__dict__), parameters, _root=True)
  File "C:\Users\my_user\AppData\Local\Programs\Python\Python35\Lib\typing.py", line 512, in __new__
    for t2 in all_params - {t1} if not isinstance(t2, TypeVar)):
  File "C:\Users\my_user\AppData\Local\Programs\Python\Python35\Lib\typing.py", line 512, in <genexpr>
    for t2 in all_params - {t1} if not isinstance(t2, TypeVar)):
  File "C:\Users\my_user\AppData\Local\Programs\Python\Python35\Lib\typing.py", line 1077, in __subclasscheck__
    if super().__subclasscheck__(cls):
  File "c:\my_project\venv\lib\abc.py", line 225, in __subclasscheck__
    for scls in cls.__subclasses__():
TypeError: descriptor '__subclasses__' of 'type' object needs an argument
```
#### Expected Output
No error
Currently works on pandas==0.24.2

#### Output of ``pd.show_versions()``
https://files.pythonhosted.org/packages/19/b6/124914598447928889c6fe08a3e40ff791e1c66c7f6e3234374eefe38359/pandas-0.25.0rc0-cp35-cp35m-win_amd64.whl#sha256=ef9976c835d4a3f1de54b0d176711d27952b48cdf752b37fe60e69395fe35b9b

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/27247

### CODE REVIEW

File: pandas/tests/test_downstream.py [hunk 1]
```diff
@@ -150,3 +150,11 @@ def test_missing_required_dependency():
     output = exc.value.stdout.decode()
     for name in ["numpy", "pytz", "dateutil"]:
         assert name in output
+
+
+def test_old_python_raises():
+    code = "import sys; sys.version_info = (3, 5, 2); import pandas"
+    # match = 'pandas requires Python >=3.5.3 but 3.5.2 is installed'
+    out = subprocess.run(["python", "-c", code], check=False, capture_output=True)
```
Comment (WillAyd) [inline pandas/tests/test_downstream.py]: capture_output I don't think is available until 3.7 so source of CI errors. Maybe not needed at all?

File: pandas/tests/test_downstream.py [hunk 2]
```diff
@@ -150,3 +150,11 @@ def test_missing_required_dependency():
     output = exc.value.stdout.decode()
     for name in ["numpy", "pytz", "dateutil"]:
         assert name in output
+
+
+def test_old_python_raises():
```
Comment (jreback) [inline pandas/tests/test_downstream.py]: we *could* move the import of just ``pandas.compat`` at the beginning of init and define PY35 there (and import here); would put the code more logically together.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by shoyer** [comment]
Would it work to put this error in setup.py instead? Install time errors seem better than import time errors. Or are there cases where that doesn’t get run?

**Comment by jreback** [comment]
> Would it work to put this error in setup.py instead? Install time errors seem better than import time errors. Or are there cases where that doesn’t get run?

i think if you have a old pip and install from a wheel this wont be hit in setup.py
only at import time

**Comment by TomAugspurger** [comment]
Agreed that this should be in `setup.py`, but after thinking about it more, I'm hesitant to duplicate things that pip is doing (when used correctly).

The reported fail case in https://github.com/pandas-dev/pandas/issues/27247 was from (incorrectly) using `python setup.py develop` directly. I'd rather educate users on how pip says it should be done, rather than add duplicative checks.

**Pull request closed without merging**